client/build.rs

```diff
@@ -5,15 +5,23 @@
 //! See [cargo issue #4001](https://github.com/rust-lang/cargo/issues/4001)
 
 use core::str::from_utf8;
-use std::{env, path::Path, process::Command};
+use std::{
+    env,
+    ffi::OsStr,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+use eyre::{eyre, Context as _, Result};
 
 #[allow(clippy::expect_used)]
 fn main() {
-    const SMARTCONTRACT_LOCAL_PATH: &str =
-        "tests/integration/create_nft_for_every_user_smartcontract";
+    const TEST_SMARTCONTRACTS_DIR: &str = "tests/integration/smartcontracts";
+    const TEST_SMARTCONTRACTS: [&str; 2] = ["create_nft_for_every_user_smartcontract", "mint_rose"];
```

Comment:
Maybe instead of hardcoding every smartcontract directory we could iterate over them inside root smartcontracts directory? 

`Path::read_dir` could help here.